src/jni_core/src/jvm_runtime.rs

```diff
@@ -16,86 +16,106 @@ use core::option::Option::Some;
 use std::ffi::c_void;
 use std::fs;
 use std::path::Path;
-use std::sync::LazyLock;
+use std::sync::OnceLock;
 
+use jni::objects::JValueOwned;
 use jni::strings::JNIString;
 use jni::{InitArgsBuilder, JNIVersion, JavaVM, NativeMethod};
 use risingwave_common::error::{ErrorCode, RwError};
 use risingwave_common::util::resource_util::memory::system_memory_available_bytes;
 
-pub static JVM: LazyLock<Result<JavaVM, RwError>> = LazyLock::new(|| {
-    let libs_path = if let Ok(libs_path) = std::env::var("CONNECTOR_LIBS_PATH") {
-        libs_path
-    } else {
-        return Err(ErrorCode::InternalError(
-            "environment variable CONNECTOR_LIBS_PATH is not specified".to_string(),
-        )
-        .into());
-    };
+pub static JVM: JavaVmWrapper = JavaVmWrapper::new();
 
-    let dir = Path::new(&libs_path);
+pub struct JavaVmWrapper(OnceLock<Result<JavaVM, RwError>>);
 
-    if !dir.is_dir() {
-        return Err(ErrorCode::InternalError(format!(
-            "CONNECTOR_LIBS_PATH \"{}\" is not a directory",
-            libs_path
-        ))
-        .into());
+impl JavaVmWrapper {
+    const fn new() -> Self {
+        Self(OnceLock::new())
     }
 
-    let mut class_vec = vec![];
+    pub fn get(&self) -> Option<&Result<JavaVM, RwError>> {
+        self.0.get()
+    }
 
-    if let Ok(entries) = fs::read_dir(dir) {
-        for entry in entries.flatten() {
-            let entry_path = entry.path();
-            if entry_path.file_name().is_some() {
-                let path = std::fs::canonicalize(entry_path)?;
-                class_vec.push(path.to_str().unwrap().to_string());
-            }
-        }
-    } else {
-        return Err(ErrorCode::InternalError(format!(
-            "failed to read CONNECTOR_LIBS_PATH \"{}\"",
-            libs_path
-        ))
-        .into());
+    pub fn get_or_init(&self) -> Result<&JavaVM, &RwError> {
+        self.0.get_or_init(Self::inner_new).as_ref()
```

Comment:
nit: What about `get_or_try_init` here? So that the failure will not be "permanent".

Reply:
It looks like the JVM initialization failure would be permanent unless we restart and provide the correct configuration like `CONNECTOR_LIBS_PATH`.🤔

Reply:
Okay, then that makes sense.